core/src/labels/labels.cpp

```diff
@@ -169,187 +174,135 @@ void Labels::skipTransitions(const std::vector<std::unique_ptr<Style>>& _styles,
     }
 }
 
-void Labels::checkRepeatGroups(std::vector<Label*>& _visibleSet) const {
+bool Labels::labelComparator(const LabelEntry& _a, const LabelEntry& _b) {
+    if (_a.proxy != _b.proxy) {
+        return _b.proxy;
+    }
+
+    if (_a.priority != _b.priority) {
+        return _a.priority < _b.priority;
+    }
+
+    auto l1 = _a.label;
+    auto l2 = _b.label;
+
+    // Note: This causes non-deterministic placement, i.e. depending on
+    // navigation history.
+    if (l1->occludedLastFrame() != l2->occludedLastFrame()) {
```

Comment:
It's not 'needed' to do this. It's a matter of preference of less flickering vs having always the same set of labels shown for a viewport - This is the behavior we had before so I kept it and added the Note.